packages/terra-date-picker/src/react-datepicker/calendar.jsx

```diff
@@ -99,7 +99,7 @@ export default class Calendar extends React.Component {
     /**
      * Highlight range of dates with custom classes.
      */
-    highlightDates: PropTypes.instanceOf(Map),
+    highlightDates: PropTypes.instanceOf(WeakMap),
```

Comment:
Did this play a role in contributing to the memory leak? Or is it just better in general to not keep a strong reference to the map keys?

Reply:
No, but This change was done more as precaution of preventing possible leaks that could happen in some edge cases with key objects.